pkg/api/query/querypb/query.proto

```diff
@@ -41,6 +41,7 @@ message QueryRequest {
   ShardInfo shard_info = 11;
 
   int64 lookback_delta_seconds = 12;
+  string engine = 13;
```

Comment:
Maybe using an enum instead of a string here would be better?

Something like,
```
enum EngineType {
  THANOS = 0;
  PROMETHEUS = 1;
}
```

Reply:
I am not sure how to hamdle default case with this?

Reply:
Why not? Would be the same as erroring out in the HTTP API, no? 🙂

Reply:
For [enums](https://protobuf.dev/programming-guides/proto3/#enum), the default value is the first defined enum value, which must be 0. THANOS in this case.

https://protobuf.dev/programming-guides/proto3/#default

Reply:
Ah yes, it is. I was talking about how to handle `default` in gRPC API  switch case

Reply:
the enum by default choose prometheus if not provided. where in string a empty string is form and we can use `defaultEngine` value if empty.

Does that mean to add a default `engineType` in enum?